src/server.rs

```diff
@@ -297,39 +281,44 @@ impl SubstrateService {
     }
 
     #[tool(
-        description = "Query and filter blockchain events within a specified block range. Supports filtering by pallet and event name with partial matching. Use this to find specific events like transfers, staking rewards, or governance actions."
+        description = "Query events from blocks. Supports querying by pallet and event name. Returns event details such as pallet name, event index and data. Supports relative block numbers (e.g., -10 for 10 blocks ago). If to_block is left blank, will query only a single block equal to from_block; to query a range it needs both parameters. Maximum block range is 100 blocks."
     )]
     pub async fn query_events(
         &self,
-        Parameters(args): Parameters<EventFilterArgs>,
+        Parameters(args): Parameters<QueryEventsProperties>,
     ) -> Result<CallToolResult, McpError> {
         // Connect to the chain using subxt
         let client = OnlineClient::<PolkadotConfig>::from_url(&args.rpc_url)
             .await
             .map_err(|e| McpError {
                 code: rmcp::model::ErrorCode(-32603),
-                message: format!("Failed to connect to chain: {e}").into(),
+                message: format!(
+                    "Failed to connect to chain with URL '{}': {e}",
+                    args.rpc_url
```

Comment:
Just stylistic, but I wouldn't mismatch formatting styles without a good reason to do so :)